File: task_equivalents/llm_translations/gemini/base/python/statistical_modelling_analysis/01_linear_regression_inference.py
import pandas as pd
import statsmodels.formula.api as smf

auto = pd.read_csv('data/csv/Auto.csv', na_values='?')
auto = auto.dropna()

model1 = smf.ols('mpg ~ horsepower', data=auto).fit()
print(model1.summary())

new_data = pd.DataFrame({'horsepower': [98]})
predictions = model1.get_prediction(new_data)
summary_frame = predictions.summary_frame(alpha=0.05)

# Confidence interval
print("Confidence Interval:")
print(summary_frame[['mean', 'mean_ci_lower', 'mean_ci_upper']])

# Prediction interval
print("\nPrediction Interval:")
print(summary_frame[['mean', 'obs_ci_lower', 'obs_ci_upper']])

File: task_equivalents/llm_translations/gemini/base/r/statistical_modelling_analysis/01_linear_regression_inference.R
Auto <- read.csv('data/csv/Auto.csv', na.strings = '?')
Auto <- na.omit(Auto)

model1_fit <- lm(mpg ~ horsepower, data = Auto)
print(summary(model1_fit))

new_data <- data.frame(horsepower = 98)

conf_interval <- predict(model1_fit, new_data, interval = "confidence", level = 0.95)
print(conf_interval)

pred_interval <- predict(model1_fit, new_data, interval = "prediction", level = 0.95)
print(pred_interval)